src/sc2api/sc2_agent.cc

```diff
@@ -122,66 +128,124 @@ void ActionImp::SendChat(const std::string& message, ChatChannel channel) {
+    UnitCommand(tags, ability, point, queued_command);
+}
 
+void ActionImp::UnitCommand(const Units& units, AbilityID ability, const Unit* target, bool queued_command) {
+    std::vector<Tag> tags;
     for (auto unit : units) {
         if (!unit) continue;
-        unit_command->add_unit_tags(unit->tag);
+        tags.push_back(unit->tag);
     }
+    UnitCommand(tags, ability, target->tag, queued_command);
 }
 
-void ActionImp::UnitCommand(const Units& units, AbilityID ability, const Point2D& point, bool queued_command) {
+void ActionImp::UnitCommand(Tag tag, AbilityID ability, bool queued_command) {
     SC2APIProtocol::RequestAction* request_action = GetRequestAction();
     SC2APIProtocol::Action* action = request_action->add_actions();
     SC2APIProtocol::ActionRaw* action_raw = action->mutable_action_raw();
-    SC2APIProtocol::ActionRawUnitCommand* unit_command = action_raw->mutable_unit_command();
+    SC2APIProtocol::ActionRawUnitCommand* tag_command = action_raw->mutable_unit_command();
 
-    unit_command->set_ability_id(ability);
-    SC2APIProtocol::Point2D* target_point = unit_command->mutable_target_world_space_pos();
+    tag_command->set_ability_id(ability);
+    tag_command->set_queue_command(queued_command);
+    tag_command->add_unit_tags(tag);
+}
+
+void ActionImp::UnitCommand(Tag tag, AbilityID ability, const Point2D& point, bool queued_command) {
+    SC2APIProtocol::RequestAction* request_action = GetRequestAction();
+    SC2APIProtocol::Action* action = request_action->add_actions();
+    SC2APIProtocol::ActionRaw* action_raw = action->mutable_action_raw();
+    SC2APIProtocol::ActionRawUnitCommand* tag_command = action_raw->mutable_unit_command();
+
+    tag_command->set_ability_id(ability);
+    SC2APIProtocol::Point2D* target_point = tag_command->mutable_target_world_space_pos();
     target_point->set_x(point.x);
     target_point->set_y(point.y);
-    unit_command->set_queue_command(queued_command);
+    tag_command->set_queue_command(queued_command);
+    tag_command->add_unit_tags(tag);
+}
 
-    for (auto unit : units) {
-        if (!unit) continue;
-        unit_command->add_unit_tags(unit->tag);
-    }
+void ActionImp::UnitCommand(Tag tag, AbilityID ability, const Tag target_tag, bool queued_command) {
+    SC2APIProtocol::RequestAction* request_action = GetRequestAction();
+    SC2APIProtocol::Action* action = request_action->add_actions();
+    SC2APIProtocol::ActionRaw* action_raw = action->mutable_action_raw();
+    SC2APIProtocol::ActionRawUnitCommand* tag_command = action_raw->mutable_unit_command();
+
+    tag_command->set_ability_id(ability);
+    tag_command->set_target_unit_tag(target_tag);
+    tag_command->set_queue_command(queued_command);
+    tag_command->add_unit_tags(tag);
 }
 
-void ActionImp::UnitCommand(const Units& units, AbilityID ability, const Unit* target, bool queued_command) {
+void ActionImp::UnitCommand(const std::vector<Tag>& tags, AbilityID ability, bool queued_command) {
```

Comment:
I think we should introduce `Tags` entity (e.g. by using typedef) similarly to `Units` and use it everywhere.